### PR TITLE
EWPP-63: Theme "Call for tender" full view

### PR DESCRIPTION
**Pull request by yenyasinn**
### Description

TenderNodeWrapper has been added.

### CODE REVIEW

File: modules/oe_content_tender/tests/src/Kernel/TenderKernelTestBase.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+   * @var array
+   */
+  public static $modules = [
+    'field',
+    'field_group',
+    'datetime_range',
+    'entity_reference_revisions',
+    'link',
+    'image',
+    'inline_entity_form',
```
Comment (drishu) [inline modules/oe_content_tender/tests/src/Kernel/TenderKernelTestBase.php:26]: there are some modules here which are not needed, like inline_entity_form, lets remove them one by one to see which are actually needed, the less is loaded the better

**Review by drishu — CHANGES_REQUESTED** [review]

File: modules/oe_content_tender/src/TenderNodeWrapperInterface.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+/**
+ * Interface for the tender entity wrapper.
+ */
+interface TenderNodeWrapperInterface {
+
+  /**
+   * Tender isn't available. Is used when opening date isn't defined.
+   */
+  const TENDER_STATUS_NOT_AVAILABLE = 'n/a';
```
Comment (ademarco) [inline modules/oe_content_tender/src/TenderNodeWrapperInterface.php:18]:
```suggestion
  const TENDER_STATUS_NOT_AVAILABLE = 'not_available';
```
So we don't have `/` constant values.

File: modules/oe_content_tender/src/TenderNodeWrapperInterface.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+   */
+  public function isClosed(): bool;
+
+  /**
+   * Check whether the tender status is not available.
+   *
+   * @return bool
+   *   Whereas the tender status is not available.
+   */
+  public function isNotAvailable(): bool;
```
Comment (ademarco) [inline modules/oe_content_tender/src/TenderNodeWrapperInterface.php:65]: Given that the status field is optional (can't recall right now) let's change this into:
```suggestion
  public function hasStatus(): bool;
```

File: modules/oe_content_tender/src/TenderNodeWrapperInterface.php [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+   */
+  public function hasDeadlineDate(): bool;
+
+  /**
+   * Gets status of the tender.
+   *
+   * @return string
+   *   Tender status.
+   */
+  public function getTenderStatus(): string;
```
Comment (ademarco) [inline modules/oe_content_tender/src/TenderNodeWrapperInterface.php:105]:
```suggestion
  public function getStatus(): string;
```

**Review by ademarco — CHANGES_REQUESTED** [review]

**Review by ademarco — APPROVED** [review]

**Review by drishu — APPROVED** [review]

File: modules/oe_content_tender/tests/src/Kernel/TenderNodeWrapperTest.php [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    TenderNodeWrapper::getInstance($entity);
+  }
+
+  /**
+   * Test exception thrown if an unsupported entity bundle is passed.
+   */
+  public function testUnsupportedEntityBundle(): void {
+    $this->expectException(\InvalidArgumentException::class);
+    $this->expectExceptionMessage("The current wrapper only accepts 'node' entities of type 'oe_tender'.");
+    $node_type = NodeType::create([
```
Comment (22Alexandra) [inline modules/oe_content_tender/tests/src/Kernel/TenderNodeWrapperTest.php:39]:
```suggestion
NodeType::create([
      'type' => 'test',
      'name' => 'Test bundle',
    ])->save();
```

Comment (22Alexandra) [reply modules/oe_content_tender/tests/src/Kernel/TenderNodeWrapperTest.php:39]: $node_type variable is not used.

Comment (yenyasinn) [reply modules/oe_content_tender/tests/src/Kernel/TenderNodeWrapperTest.php:39]: $node_type is used on the line 43.

Comment (22Alexandra) [reply modules/oe_content_tender/tests/src/Kernel/TenderNodeWrapperTest.php:39]: Indeed, but you can save after closing the bracket, no?

File: modules/oe_content_tender/tests/src/Kernel/TenderNodeWrapperTest.php [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+      'type' => 'test',
+      'name' => 'Test bundle',
+    ]);
+    $node_type->save();
+    $entity = Node::create([
+      'type' => 'test',
+    ]);
```
Comment (22Alexandra) [inline modules/oe_content_tender/tests/src/Kernel/TenderNodeWrapperTest.php:46]: Also, this should be saved. Currently, is only created.

Comment (ademarco) [reply modules/oe_content_tender/tests/src/Kernel/TenderNodeWrapperTest.php:46]: This is actually ok, here we are testing unsupported bundle.

**Review by 22Alexandra — APPROVED** [review]

**Pull request merged**